pkg/apis/clusteroperator/types.go

```diff
@@ -151,6 +166,17 @@ type ClusterStatus struct {
 	// For machine set hardware, see the status of each machine set resource.
 	Provisioned bool
 
+	// ProvisioningJobGeneration is the generation of the cluster resource used to
+	// launch the last provisioning job.
+	// +optional
```

Comment:
I am thinking that this should not be `+optional`. Even when a provisioning job has not been started yet, you could theoretically consider that the generation that was provisioned is the 0 generation. If it is optional, then the kube API conventions stipulate that the field should be a `*int64`.